### PR TITLE
Improve snap logs

### LINKED ISSUE #3742: Snap sync stall
### Description
Tried a snap sync with bonsai storage on MainNet. It wound up stalling and not making any progress for many hours.   Most likely relevant section of the logs is below - it was healing the state then repivoted and that's the last I heard of any sync progress.

I've restarted it and it's now back to downloading world state nodes and importing blocks.

This is on the `dev-teku-besu-ohio-mainnet-shadow-2` box for those with ConsenSys AWS access - feel free to poke around on there or I'm happy to grab more detailed logs etc.

Besu version: besu/v22.4.0-RC2-dev-24e417d4/linux-x86_64/corretto-java-17

```
2022-04-20 00:34:44.139+00:00 | EthScheduler-Services-13 (batchPersistBigStorageData) | INFO  | SnapWorldDownloadState | Retrieved 1164207491 world state nodes
2022-04-20 00:34:46.143+00:00 | EthScheduler-Services-13 (batchPersistBigStorageData) | INFO  | SnapWorldDownloadState | Retrieved 1164409126 world state nodes
2022-04-20 00:34:48.386+00:00 | EthScheduler-Services-8 (batchPersistStorageData) | INFO  | SnapWorldDownloadState | Retrieved 1164609130 world state nodes
2022-04-20 00:34:49.935+00:00 | EthScheduler-Services-3 (batchPersistAccountData) | INFO  | SnapWorldDownloadState | Retrieved 1164818343 world state nodes
2022-04-20 00:34:54.084+00:00 | EthScheduler-Services-18 (batchPersistCodeData) | INFO  | SnapWorldDownloadState | Retrieved 1165018343 world state nodes
2022-04-20 00:34:57.751+00:00 | EthScheduler-Services-8 (batchPersistStorageData) | INFO  | SnapWorldDownloadState | Retrieved 1165220762 world state nodes
2022-04-20 00:34:59.204+00:00 | EthScheduler-Services-13 (batchPersistBigStorageData) | INFO  | SnapWorldDownloadState | Retrieved 1165426568 world state nodes
2022-04-20 00:35:01.168+00:00 | EthScheduler-Services-13 (batchPersistBigStorageData) | INFO  | SnapWorldDownloadState | Retrieved 1165649698 world state nodes
2022-04-20 00:35:03.393+00:00 | EthScheduler-Services-28 (requestCompleteTask) | INFO  | SnapWorldDownloadState | Starting world state heal process from peers (generated nodes 1165819046)
2022-04-20 00:35:13.896+00:00 | Timer-0 | INFO  | DNSTimerTask | Refreshing DNS records with enrtree://AKA3AM6LPBYEUDMVNU3BSVQJ5AD45Y7YPOHJLEF6W26QOE4VTUDPE@all.mainnet.ethdisco.net
2022-04-20 00:35:20.113+00:00 | EthScheduler-Services-26 (batchPersistTrieNodeData) | INFO  | SnapWorldDownloadState | Healed 10000 world sync nodes
2022-04-20 00:35:25.279+00:00 | vert.x-eventloop-thread-0 | WARN  | EngineExchangeTransitionConfiguration | not called in 120 seconds, consensus client may not be connected
2022-04-20 00:35:30.348+00:00 | EthScheduler-Services-26 (batchPersistTrieNodeData) | INFO  | SnapWorldDownloadState | Healed 20002 world sync nodes
2022-04-20 00:35:41.716+00:00 | EthScheduler-Services-26 (batchPersistTrieNodeData) | INFO  | SnapWorldDownloadState | Healed 30003 world sync nodes
2022-04-20 00:36:01.800+00:00 | EthScheduler-Services-26 (batchPersistTrieNodeData) | INFO  | SnapWorldDownloadState | Healed 40006 world sync nodes
2022-04-20 00:36:11.851+00:00 | EthScheduler-Services-26 (batchPersistTrieNodeData) | INFO  | SnapWorldDownloadState | Healed 50007 world sync nodes
2022-04-20 00:36:13.896+00:00 | Timer-0 | INFO  | DNSTimerTask | Refreshing DNS records with enrtree://AKA3AM6LPBYEUDMVNU3BSVQJ5AD45Y7YPOHJLEF6W26QOE4VTUDPE@all.mainnet.ethdisco.net
2022-04-20 00:36:24.541+00:00 | EthScheduler-Services-26 (batchPersistTrieNodeData) | INFO  | SnapWorldDownloadState | Healed 60010 world sync nodes
2022-04-20 00:36:25.867+00:00 | EthScheduler-Services-24 (checkNewPivotBlock) | INFO  | DynamicPivotBlockManager | Select new pivot block 14618744 0x36d62464ff650fb2ac5b725918e227eae769b1347a5239b71a15f622baa8ebd0
2022-04-20 00:37:13.896+00:00 | Timer-0 | INFO  | DNSTimerTask | Refreshing DNS records with enrtree://AKA3AM6LPBYEUDMVNU3BSVQJ5AD45Y7YPOHJLEF6W26QOE4VTUDPE@all.mainnet.ethdisco.net
2022-04-20 00:37:25.379+00:00 | vert.x-eventloop-thread-0 | WARN  | EngineExchangeTransitionConfiguration | not called in 120 seconds, consensus client may not be connected
2022-04-20 00:38:13.896+00:00 | Timer-0 | INFO  | DNSTimerTask | Refreshing DNS records with enrtree://AKA3AM6LPBYEUDMVNU3BSVQJ5AD45Y7YPOHJLEF6W26QOE4VTUDPE@all.mainnet.ethdisco.net
2022-04-20 00:39:13.896+00:00 | Timer-0 | INFO  | DNSTimerTask | Refreshing DNS records with enrtree://AKA3AM6LPBYEUDMVNU3BSVQJ5AD45Y7YPOHJLEF6W26QOE4VTUDPE@all.mainnet.ethdisco.net
2022-04-20 00:39:25.478+00:00 | vert.x-eventloop-thread-0 | WARN  | EngineExchangeTransitionConfiguration | not called in 120 seconds, consensus client may not be connected
2022-04-20 00:40:13.896+00:00 | Timer-0 | INFO  | DNSTimerTask | Refreshing DNS records with enrtree://AKA3AM6LPBYEUDMVNU3BSVQJ5AD45Y7YPOHJLEF6W26QOE4VTUDPE@all.mainnet.ethdisco.net
2022-04-20 00:41:13.896+00:00 | Timer-0 | INFO  | DNSTimerTask | Refreshing DNS records with enrtree://AKA3AM6LPBYEUDMVNU3BSVQJ5AD45Y7YPOHJLEF6W26QOE4VTUDPE@all.mainnet.ethdisco.net
2022-04-20 00:41:25.552+00:00 | vert.x-eventloop-thread-0 | WARN  | EngineExchangeTransitionConfiguration | not called in 120 seconds, consensus client may not be connected
```

### LINKED ISSUE #3744: Improve logging of snap sync progress
### Description
Currently snap sync just logs a lot of:
```
SnapWorldDownloadState | Retrieved 804128239 world state nodes
```

which is overly noisy in the logs and doesn't tell me anything useful.  One of the advantages of snap sync over fast sync is that you should be able to provide a reasonable estimate of the percentage of world state download you've completed.  Ideally we'd log a message every 5-10 seconds or so that shows the current percentage complete (and possibly how many state nodes were downloaded in that time but it's not particularly interesting for the user).

### PR DESCRIPTION
**Pull request by matkt**
Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

## PR description

This PR will significantly improve the snapsync logs by adding more statistics
```
Snapsync in progress synced=17.66%, accounts=1757406, slots=17596376, codes=628631, nodes=26844745
```

## Fixed Issue(s)

fixes #3744
fixes #3742 

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by frankisawesome** [comment]
It would be good practice to seperate the bug fix code changes from improvements

**Review by frankisawesome — COMMENTED** [review]
Can we add a test to recreate the bug that caused https://github.com/hyperledger/besu/issues/3742 ?

**Review by frankisawesome — APPROVED** [review]
The logging changes looks good, I don't know enough about snapsync to review https://github.com/hyperledger/besu/pull/3761

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/StackTrie.java [hunk 1]
```diff
@@ -59,10 +61,12 @@ public StackTrie(
     this.maxSegments = maxSegments;
     this.startKeyHash = startKeyHash;
     this.elements = new LinkedHashMap<>();
+    this.nbElements = new AtomicLong();
```
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/StackTrie.java:64]: s/nbElements/numberOfElements 
or elementsCount ?

**Review by macfarla — APPROVED** [review]

**Comment by matkt** [comment]
I just removed the heal fix. And I created another for this part

**Review by macfarla — APPROVED** [review]

**Pull request merged**